lib/fetching/content-services/facebook-content-service.js

```diff
@@ -26,116 +27,139 @@ FacebookContentService.prototype._doFetch = function(options, callback) {
+    });
     callback([]);
     return;
   }
 
-  var queries = {};
+  var source = this._url.trim();
 
-  // Determine Source ID of object
-  queries.source = 'SELECT id, site, type, url FROM object_url WHERE url = "' + this._url + '"';
+  // Parse out source name from url
+  if (source.indexOf('?') === -1) {
+    source = source.split('?')[0];
+  }
 
-  // Fetch posts and comments
-  queries.posts = 'SELECT source_id, post_id, updated_time, created_time, actor_id, permalink, message';
-  queries.posts += ' FROM stream';
-  queries.posts += ' WHERE source_id IN (SELECT id FROM #source)';
-  // New comments trigger a fresh updated_time in its parent post
-  if (this._lastReportDate) queries.posts += ' AND updated_time > ' + (this._lastReportDate.getTime()/1000);
-  queries.posts += ' ORDER BY updated_time';
+  if (source.lastIndexOf('/') === source.length - 1) {
+    source = source.substring(0, source.length - 1);
+  }
 
-  // Get new comments associated with posts.
-  queries.comments = "SELECT id, fromid, time, text, post_id FROM comment WHERE (post_id IN (SELECT post_id FROM #posts))";
-  if (this._lastReportDate) queries.comments += ' AND time > ' + (this._lastReportDate.getTime()/1000);
-  queries.comments += ' ORDER BY time';
+  source = source.substring(source.lastIndexOf('/') + 1, source.length);
 
-  queries.authors = 'SELECT id, name FROM profile WHERE id IN (SELECT actor_id FROM #posts) OR id IN (SELECT fromid FROM #comments)';
 
-  this._doRequest(queries, function(err, res) {
+  this._doRequest(source, function(err, data) {
     if (err) {
-      self.emit('error', new Error(err.message))
+      self.emit('error', new Error(err.message));
       return callback([]);
     }
+    callback(self._handleResults(data, source));
+  })
+};
 
-    // This will be an array of info about the source.
-    var sourceInfo = _.findWhere(res.data, {name: 'source'}).fql_result_set;
+FacebookContentService.prototype._doRequest = function (source, callback) {
+  var self = this;
+  var data = [];
+  var params = {};
 
-    if (sourceInfo.length === 0) {
-      self.emit('error', new Error('Invalid Facebook URL'));
-      callback([]);
-    } else {
-      var posts = _.findWhere(res.data, {name: 'posts'}).fql_result_set;
-      var comments = _.findWhere(res.data, {name: 'comments'}).fql_result_set;
-      var authors = _.findWhere(res.data, {name: 'authors'}).fql_result_set;
-      callback(self._handleResults(posts, comments, authors));
-    }
-  });
+  if (this._lastReportDate) {
+    params.since = this._lastReportDate.getTime() / 1000;
+  }
+  // Grab posts off of sources feed
+  self._doGet('/v2.2/' + source + '/feed', params, data, callback);
+
+  //If no new posts, grab comments off of most recent post
+  if((!data || data.length == 0) && this.posts[source] != null) {
+    self._doGet('/v2.2/' + this.posts[source].id + '/comments?order=reverse_chronological', params, data, callback);
```

Comment:
I just tested this and I think in fact p1 is returned.

Test FB group: https://www.facebook.com/groups/539367472907722/

Graph API request: https://developers.facebook.com/tools/explorer/145634995501895/?method=GET&path=539367472907722%2Ffeed%3Fsince%3D1454685714&version=v2.5

Note how the first post is retrieved while the second one is not. This is because the first post has a comment after the `since` date.

The docs also say the API can return 100 posts at a time. That's not bad in terms of quantity. The tricky part will be fetching the comments. If there were some way to know if a post has new comments, that would save some requests. Perhaps you can look into this.